x/stakeibc/keeper/ibc_handlers.go

```diff
@@ -177,9 +177,8 @@ func (k Keeper) HandleAcknowledgement(ctx sdk.Context, modulePacket channeltypes
 		}
 	}
 
-	if recordIdToDelete >= 0 {
-		k.RecordsKeeper.RemoveDepositRecord(ctx, cast.ToUint64(recordIdToDelete))
-	}
+	// no need to check id because error case is handled above
+	k.RecordsKeeper.RemoveDepositRecord(ctx, cast.ToUint64(recordIdToDelete))
```

Comment:
flagging

Reply:
Addressed per new cast fork and using cast.To____E